gc.c

```diff
@@ -7208,6 +7208,47 @@ gc_mark_imemo(rb_objspace_t *objspace, VALUE obj)
     }
 }
 
+static bool
+gc_declarative_marking_p(const rb_data_type_t *type)
+{
+    return (type->flags & RUBY_TYPED_DECL_MARKING) != 0;
+}
+
+static inline void
+gc_mark_from_offset(rb_objspace_t *objspace, VALUE *data_struct, VALUE offset)
+{
+    VALUE markable = *(data_struct + (offset / sizeof(VALUE)));
+    rb_gc_mark_movable(markable);
+}
```

Comment:
- Use `size_t` for offsets.
- Use `EDGE` macro also in this function.

```suggestion
#define EDGE *(VALUE *)((char *)data_struct + offset)

static inline void
gc_mark_from_offset(rb_objspace_t *objspace, void *data_struct, size_t offset)
{
    rb_gc_mark_movable(EDGE);
}
```